### PR TITLE
Cloud layer volume override

### PR DESCRIPTION
**Pull request by adrien-de-tocqueville**
**Guide** : https://github.com/Unity-Technologies/Graphics/blob/master/.github/pr-read.png.md

**Display Addon** : https://userstyles.org/styles/182991/unity-graphics-pr-readme

# Purpose of this PR

> Why is this PR needed, what hard problem is it solving/fixing?

This is the second part of the cloud layer (as designed [here](https://docs.google.com/document/d/1ITVvxtJiI244DrX7Vep_LMEiusiafVHG_m8KtpVewow/edit), first part [there](https://github.com/Unity-Technologies/Graphics/pull/44)).

This PR is adding a new volume override in Sky > Cloud Layer, which displays a texture on top of any sky currently active. The settings are mostly the same as those for the HDRISky flowmap

cloudlayer

Example with the PBR sky
pbr

# Testing status
* Built a player
* Tested missing doc
* Tested with all 3 types of sky

No automated test added

## Links
**Yamato**: (Select your branch) https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fcloud-layer
> Any test projects or documents to go with this to help reviewers?

Design doc: https://docs.google.com/document/d/1ITVvxtJiI244DrX7Vep_LMEiusiafVHG_m8KtpVewow/edit

# Comments to reviewers
> Notes for the reviewers you have assigned.

The Cloud Layer should also impact the static lighting sky. I am not very familiar with this feature so there's maybe something incorrect there

### CODE REVIEW

**Comment by iM0ve** [comment]
**Issues**

1) Noticed when adding disabled Cloud override, it affects the fog in some way. Is it intended?

change

**Comment by SeanPuller** [comment]
I see some blending issues with HDRi's

image

File: com.unity.render-pipelines.high-definition/Runtime/Sky/HDRISky/HDRISky.shader [hunk 1]
```diff
@@ -122,7 +126,7 @@ Shader "Hidden/HDRP/Sky/HDRISky"
     {
         const float alpha = (_GroundLevel - _WorldSpaceCameraPos.y)/dir.y;
 
-        return _WorldSpaceCameraPos + alpha*dir;
+        return (float3)_WorldSpaceCameraPos + alpha*dir;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/HDRISky/HDRISky.shader:129]: (float3)_WorldSpaceCameraPos  isn't required anymore (see latest staging), please remove it at all location

**Comment by sebastienlagarde** [comment]
@iM0ve have you check the baked sky (static lighting) ?

**Comment by iM0ve** [comment]
On my end issue still seems to be present. Baking when layer is off, still bakes the clouds.

0CyAvSAfBa

**Comment by SeanPuller** [comment]
Works perfectly with the HDRi and Gradient skies with a intensity multipier of 1. With the procedural sky the cloud layer isn't visible until the intensity value is set to 1000 and looks as expected with 10000. This will be hard to discover. It will be very valuable to include this in the documentation and address this in the future.

**Comment by iM0ve** [comment]
Baking issue is fixed. Found a new one.
Clouds provide specular lighting just as reflection probes. However, the Indirect Lighting Controller only works on Reflection probes (see first part of gif), and has no effect on the clouds produced specular lighting (second part of the gif)

3dCojjx7o5
Above gif is using Debug->Lighting-> Debug Mode -> Specular lighting

**Pull request merged**